File: java/src/main/java/com/svix/agentic_commerce_protocol/CheckoutSessionStatus.java
// This file is @generated
package com.svix.agentic_commerce_protocol;

import com.fasterxml.jackson.annotation.JsonValue;
import com.svix.agentic_commerce_protocol.Utils.ToQueryParam;

public enum CheckoutSessionStatus implements ToQueryParam {
    NOT_READY_FOR_PAYMENT("not_ready_for_payment"),
    READY_FOR_PAYMENT("ready_for_payment"),
    COMPLETED("completed"),
    CANCELED("canceled"),
    IN_PROGRESS("in_progress");
    private final String value;

    CheckoutSessionStatus(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: kotlin/src/main/kotlin/com/svix/agentic_commerce_protocol/CheckoutSessionStatus.kt
// This file is @generated
package com.svix.agentic_commerce_protocol

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.serialization.json.jsonPrimitive

@Serializable
enum class CheckoutSessionStatus : ToQueryParam {
    @SerialName("not_ready_for_payment") NOT_READY_FOR_PAYMENT,
    @SerialName("ready_for_payment") READY_FOR_PAYMENT,
    @SerialName("completed") COMPLETED,
    @SerialName("canceled") CANCELED,
    @SerialName("in_progress") IN_PROGRESS;

    override fun toQueryParam() = Json.encodeToJsonElement(this).jsonPrimitive.content
}
